tests/other/metadce/hello_libcxx_O2.size

```diff
@@ -1 +1 @@
-124705
+124939
```

Comment:
Why does code size change here, as this is not using `EXIT_RUNTIME`? (as you said in the PR description, I agree that size should only change when that is enabled)

Reply:
Thanks for pointing that out.  I turns out we would really rather use dummy functions for atexit when EXIT_RUNTIME is false.   I've updated this PR to use a dummy set of atexit functions... and I'm actually hopefully this could lead to better DCE because the optimizer should be able to see these and know that arguments to atexit are not used!  (maybe we already do this in binaryen, but we can remove that pass now I think and rely on normal DCE).